debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -889,6 +892,24 @@ public synchronized Connection connection(boolean executeOnConnect) throws SQLEx
         return conn;
     }
 
+    /**
+     * @param jdbcConnection object representing a JDBC connection
+     * @return the IP of the node the connection is made to
+     */
+    protected String getConnectedNode(Connection connection) {
+        try (Statement st = connection.createStatement()) {
+            ResultSet rs = st.executeQuery("select inet_server_addr() connected_to_host;");
```

Comment:
Is it also possible to get the process id of the walsender on that node that is servicing this connector ?